perf/reader.go

```diff
@@ -164,6 +191,8 @@ type ReaderOptions struct {
 	// Read will process data. Must be smaller than PerCPUBuffer.
 	// The default is to start processing as soon as data is available.
 	Watermark int
+
+	Attr *unix.PerfEventAttr
```

Comment:
Using `unix.PerfEventAttr` in the API creates an interesting problem. The type is `internal/unix.PerfEventAttr` which on linux is an alias to the real type in `x/sys/unix`. On other platforms (macOS, sigh) this is a stub in `internal/unix`. Trying to fill `ReaderOptions.Attr` will cause a compile time on macOS error, since `x/sys/unix` doesn't contain the type and the user can't access `internal/unix`. Maybe we don't care about this though? It's certainly confusing when looking at generated documentation.